agent/lib/kontena/service_pods/creator.rb

```diff
@@ -159,6 +159,7 @@ def ensure_image(name)
       # @param [Docker::Container] service_container
       # @return [Boolean]
       def service_uptodate?(service_container)
+        return false unless service_container.running?
```

Comment:
This kind of change to `kontena service stop+start` would cause things like the service container's weave overlay IP to change.